### PR TITLE
[1 of 2] Extract #getTaskEndpoints method to TaskUtil

### PR DESCRIPTION
**Pull request by jfurrow**
This PR extracts the endpoint logic from `MarathonTaskDetailsList#getTaskEndpoints` into a method on `TaskUtil` so it can be used elsewhere.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @philipnrmn

File: src/js/components/MarathonTaskDetailsList.js [hunk 1]
```diff
@@ -2,53 +2,29 @@ import React from 'react';
...
-    if ((task.ports == null || task.ports.length === 0) &&
-        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
-      return 'None';
+    let endpoints = TaskUtil.getTaskEndpoints(task);
+
+    let endpointURLs = endpoints.hosts.reduce(function (memo, host) {
+      return memo.concat(endpoints.ports.map(function (port) {
+        return `${host}:${port}`;
+      }));
+    }, []);
```
Comment (kennyt) [inline src/js/components/MarathonTaskDetailsList.js]: should `TaskUtil` also have something called `#getTaskEndpointURLs`

File: src/js/utils/TaskUtil.js [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
 /* eslint-enable no-unused-vars */
+
+import MarathonStore from '../stores/MarathonStore';
 import Util from './Util';
 
 const TaskUtil = {
 
+  getTaskEndpoints(task = {}) {
+    if ((task.ports == null || task.ports.length === 0) &&
+        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
```
Comment (kennyt) [inline src/js/utils/TaskUtil.js]: this is indented too far

File: src/js/components/MarathonTaskDetailsList.js (hunk 1 above)
Comment (jfurrow) [inline src/js/components/MarathonTaskDetailsList.js]: That might be a good idea, but probably for a separate PR? Or perhaps even a component. I'm just put up PR for a component which renders the endpoint links using `TaskUtil#getTaskEndpoints` in a specific way.

File: src/js/utils/TaskUtil.js (hunk 2 above)
Comment (jfurrow) [inline src/js/utils/TaskUtil.js]: Fixed!

**Comment by mlunoe** [comment]
Oh man, I have a competing PR here: https://github.com/dcos/dcos-ui/pull/683
What do you need this for?

**Comment by jfurrow** [comment]
@mlunoe We need to display the host in the `TaskTable`: https://cl.ly/0H2E1j1h3N3m/Screen%20Shot%202016-07-12%20at%2010.19.09%20AM.png

I like that we are displaying the IP even if ports are undefined. Can you make this change to my method in `TaskUtil`?

**Comment by jfurrow** [comment]
@mlunoe And I discussed this offline; I'm going to make use of his methods once they are moved into `TaskUtil`. So this can't be merged until that happens :)

**Comment by jfurrow** [comment]
This is not necessary anymore :)

**Pull request closed without merging**